src/app/effects/settings.ts

```diff
@@ -8,7 +8,6 @@ import { Actions, Effect } from '@ngrx/effects';
 import { Store } from '@ngrx/store';
 
 import { LocalStorageService } from '../services/localstorage.service';
-import { Settings } from '../models/settings';
```

Comment:
The point of using Models is to type check the code. The final object should be similar to the `Settings` model, independently of what you do in the Object assign the final Object has to be similar to the `Settings` Object.

So this `.map<Settings>(([action, store])` is making sure that the returned object is indeed a `Settings` Object. What's happening inside the `Object.assign({}, store.settings, action.payload` doesn't matter at all, as the check only cares about the final returned Object which should be of the type `Settings`.

We can build the whole application without any type checks, but it's a good practice to minimize errors.